scrypto/src/component/kv_store.rs

```diff
@@ -57,16 +57,22 @@ impl<
         let raw_bytes = ScryptoVmV1Api::kv_entry_read(handle);
 
         // Decode and create Ref
-        let substate: Option<ScryptoValue> = scrypto_decode(&raw_bytes).unwrap();
-        match substate {
-            Option::Some(value) => Some(KeyValueEntryRef::new(
+        // [0] SBOR payload prefix
+        // [1] Value kind: ENUM
+        // [2] Enum variant discriminator: SOME, NONE
+        // [3] Enum variant field size: 1 if SOME
+        match raw_bytes[2] {
```

Comment:
Looks good, I think all these `scrypto_decode` then `scrypto_encode` pattern is attributable to the need for use to these enum/tuple wrappers. This is even seen in our system layer. I think there may be a nice abstraction to allow for simple wrapping/unwrapping of sbor values. What do you think of that/where do you think we can track such a task?